### ISSUE TITLE
Grammar not acceptig valueDomainvalues starting with "_"

### ISSUE THREAD

**Issue by antonio-olleros** [sdmx-twg/vtl#405]
Currently, the grammar defines:

valueDomainValue:
    IDENTIFIER
    | INTEGER_CONSTANT
    | NUMBER_CONSTANT;

IDENTIFIER
  :
  LETTER ([A-Za-z0-9_.])*
  | DIGITS0_9 ([A-Za-z0-9_.])+
  | '\'' (.)*? '\''
  ;

Therefre, it is not posible to write hierarchy rules like: _Z = _T + A + B; but in SDMX it is very common to have codes starting with undercode (e.g., _Z for not applicable).

Grammar should be correced to allow the use of those codes.

**Comment by linardian** [comment]
Hi Antonio,
yes you are perfectly right. I am adding this issue to the #395 and #401 related with G4 and we are going to discuss it this afternoon. 
Thank you for pointing it out

**Comment by vpinna80** [comment]
I agree with this change, and I also propose to refine the IDENTIFIER rule to allow:

* single quotes inside single quotes
* SDMX identifiers without quotes (agency:dataflow(version)/query).

The query part may be dropped if you think the sdmx-to-vtl mapping spec is going to be used (today it is not).
This is my proposal (including antonio's):

```ANTLR
IDENTIFIER
  : [A-Za-z0-9_] ([A-Za-z0-9_.])*
  | ([A-Za-z0-9_.]+:)? [A-Za-z0-9_.]+ ( \([0-9._+*~]+\) )? (/[A-Za-z0-9_.]+)?
  | '\'' ( '\\\'' | [^'] )* '\''
  ;
```

**Comment by linardian** [comment]
Is this fix already in the grammar ? @vpinna80, @NicoLaval ? If not, please go ahead with Valentino's proposal. Thanks!

**Comment by linardian** [comment]
Closed with #577
